debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/ValueTypeHelper.java

```diff
@@ -0,0 +1,309 @@
+	public static final long J9ClassLargestAlignmentConstraintDouble = J9ConstantHelper.getLong(J9JavaClassFlags.class, "J9ClassLargestAlignmentConstraintDouble", 0);
+	public static final long J9ClassLargestAlignmentConstraintReference = J9ConstantHelper.getLong(J9JavaClassFlags.class, "J9ClassLargestAlignmentConstraintReference", 0);
+	private static MethodHandle getFlattenedClassCachePointer = null;
+	private static MethodHandle flattenedClassCache_offset = null;
+	private static MethodHandle flattenedClassCache_NaS = null;
+	private static MethodHandle flattenedClassCache_clazz = null;
+	private static MethodHandle flattenedClassCache_cast = null;
+	private static Class<?> flattenedClassCachePointer = null;
+	
+	private static void throwUncheckedExceptions(Throwable t) {
+		if (t instanceof Error) {
+			throw (Error)t;
+		} else if (t instanceof RuntimeException) {
+			throw (RuntimeException)t;
+		}
+	}
+	
+	private static void handleThrowable(Throwable t) throws CorruptDataException {
+		throwUncheckedExceptions(t);
+		throw new CorruptDataException(t);
+	}
+	
+	private static boolean checkIfValueTypesAreSupported() {
+		/* Older Axxon builds have builds flags in camel case, newer builds have flags capitalized */
+		if (J9ConstantHelper.getBoolean(J9BuildFlags.class, "J9VM_OPT_VALHALLA_VALUE_TYPES", false)
+			|| J9ConstantHelper.getBoolean(J9BuildFlags.class, "opt_valhallaValueTypes", false)
+		) {
+			try {
+				Lookup lookup = MethodHandles.lookup();
+				flattenedClassCachePointer = (Class<?>) Class.forName("com.ibm.j9ddr.vm29.pointer.generated.J9FlattenedClassCachePointer");
+				getFlattenedClassCachePointer = lookup.findVirtual(J9ClassPointer.class, "flattenedClassCache", MethodType.methodType(flattenedClassCachePointer));
+				flattenedClassCache_offset = lookup.findVirtual(flattenedClassCachePointer, "offset", MethodType.methodType(UDATA.class));
+				flattenedClassCache_NaS = lookup.findVirtual(flattenedClassCachePointer, "nameAndSignature", MethodType.methodType(J9ROMNameAndSignaturePointer.class));
+				flattenedClassCache_clazz = lookup.findVirtual(flattenedClassCachePointer, "clazz", MethodType.methodType(J9ClassPointer.class));
+				flattenedClassCache_cast = lookup.findStatic(flattenedClassCachePointer, "cast", MethodType.methodType(flattenedClassCachePointer, AbstractPointer.class));
+			} catch (Throwable t) {
+				/* should not happen */
+				throwUncheckedExceptions(t);
+				logger.log(Level.SEVERE, "ValueTypeHelper: failed to get ValueTypeHelper method handles", t);
+				return false;
+			}
+			return true;
+		}
+		
+		return false;
+	}
+	
+	/**
+	 * Queries if the JVM that produced the core file supports value types. This
+	 * must be called before any other ValueTypeHelper functions are called.
+	 * 
+	 * @return true if value types are supported, false otherwise
+	 */
+	public static boolean areValueTypesSupported() {
+		return areValueTypesSupported;
```

Comment:
Suggestion: create a static method that answers an instance that provides the API that is conditional.